### ISSUE TITLE
[BUG]  METHOD_CALL detecting false positive matches

### ISSUE THREAD

**Issue by Descatles** [konveyor/analyzer-lsp#827]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

v0.6.2

### Priority

Critical

### Current Behavior

Taking this project [Biblivre](https://github.com/cleydyr/biblivre) as an example, when running on rule [local-storage-00001](https://github.com/konveyor/rulesets/blob/f04484ece838f929a213bbc50284c5858d87f80c/default/generated/cloud-readiness/70-local-storage.windup.yaml#L34). Lots of function call unrelated is matched. e.g. the following two

![Image](https://github.com/user-attachments/assets/b3848392-2ba4-4e3c-9429-937126604f52)

Seems they are matching this pattern `java.nio.file.Paths.get*` but these two function call are not from package `java.nio.file.Paths`

### Expected Behavior

When matching this pattern `java.nio.file.Paths.get*`  only method call from `java.nio.file.Paths` could be matched

### How Reproducible

Always (Default)

### Steps To Reproduce

- Use this project: https://github.com/cleydyr/biblivre
- Run this rule: https://github.com/konveyor/rulesets/blob/f04484ece838f929a213bbc50284c5858d87f80c/default/generated/cloud-readiness/70-local-storage.windup.yaml#L34

I have attached my result here

[biblivre-local-storage-00001.zip](https://github.com/user-attachments/files/20200999/biblivre-local-storage-00001.zip)

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

### Anything else?

I’ve noticed that this is a common issue in method call. 
Other rule like azure-keystore-certificates-02000, java-removals-00110 and others also produce false positives. 
There’s an existing fix in the java‑analyzer‑bundle repositoryhttps://github.com/konveyor/java-analyzer-bundle/pull/100 but seems it doesn’t address every case. 
Do we have any deeper insight into the root cause of these mismatches? And any plan to fix it further?
